The remaining windows errors are probably due to an unknown character:

```bash
l=test@example.com -d project_name=some-projectmple.com
Traceback (most recent call last):
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\joran.deschamps\.conda\envs\copier\Scripts\copier.exe\__main__.py", line 7, in <module>
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\plumbum\cli\application.py", line 639, in run
    inst, retcode = subapp.run(argv, exit=False)
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\plumbum\cli\application.py", line 634, in run
    retcode = inst.main(*tailargs)
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\copier\cli.py", line 68, in _wrapper
    return method(*args, **kwargs)
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\copier\cli.py", line 291, in main
    self.parent._worker(
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\copier\main.py", line 707, in run_copy
    src_abspath = self.template_copy_root
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\functools.py", line 967, in __get__
    val = self.func(instance)
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\copier\main.py", line 680, in template_copy_root
    subdir = self._render_string(self.template.subdirectory) or ""
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\functools.py", line 967, in __get__
    val = self.func(instance)
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\copier\template.py", line 423, in subdirectory
    return self.config_data.get("subdirectory", "")
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\functools.py", line 967, in __get__
    val = self.func(instance)
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\copier\template.py", line 279, in config_data
    result = filter_config(self._raw_config)[0]
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\functools.py", line 967, in __get__
    val = self.func(instance)
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\copier\template.py", line 244, in _raw_config
    return load_template_config(conf_paths[0])
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\copier\template.py", line 101, in load_template_config
    flattened_result = lflatten(yaml.load_all(f, Loader=_Loader))
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\funcy\seqs.py", line 193, in lflatten
    return list(flatten(seq, follow))
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\funcy\seqs.py", line 184, in flatten
    for item in seq:
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\yaml\__init__.py", line 90, in load_all
    loader = Loader(stream)
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\yaml\loader.py", line 24, in __init__
    Reader.__init__(self, stream)
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\yaml\reader.py", line 85, in __init__
    self.determine_encoding()
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\yaml\reader.py", line 124, in determine_encoding
    self.update_raw()
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\site-packages\yaml\reader.py", line 178, in update_raw
    data = self.stream.read(size)
  File "C:\Users\joran.deschamps\.conda\envs\copier\lib\encodings\cp1252.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x8d in position 751: character maps to <undefined>

```

--- a/README.md
+++ b/README.md
@@ -134,13 +134,6 @@ to PyPI. (see the "deploy" step in `workflows/ci.yml`). The version number is de
 [hatch-vcs](https://github.com/ofek/hatch-vcs)... which wraps
 [setuptools-scm](https://github.com/pypa/setuptools_scm/)
 
-To auto-deploy to PyPI, you'll need to set a `TWINE_API_KEY` environment
-variable in your github repo settings.  You can get this key from your [pypi
-account](https://pypi.org/manage/account/token/).  Then add it to your github
-repository settings as a secret named `TWINE_API_KEY`. (see [github
-docs](https://docs.github.com/en/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-a-repository))
-
-(the name `TWINE_API_KEY` is specified in `workflows/ci.yml`)
 
 ```sh
 git tag -a v0.1.0 -m v0.1.0
@@ -149,6 +142,16 @@ git push --follow-tags
 # or, specify a remote:
 # git push upstream --follow-tags
 ```
+
+
+To auto-deploy to PyPI, you will need to perform two important steps:
+1. Create a `release` environment in Github
+2. Create a trusted publisher on PyPi
+
+(1) To create the Github environment, go in your Github repository to "Settings > Environemnts". Click on "new environment" and name it `release`, then click on configure. Next, in "Deployment branches", choose "Selected branches" and add a "deploment branch rule". Use `v*.*.*` as branch name patter. This allows the CI to run on Github (see `environment: release` in the "deploy" step in `workflows/ci.yml`)! 
+
+(2) Finally, connect to PyPi (you need an account). Go to your projects and click on "Publishing" (last item on the left menu). In the section "Add a new pending publisher", enter the project name (as in your `pyproject.toml`), the organization or username of the repository owner (on Github) and the repository name. Next, enter `ci.yml` as "Workflow name" and `release` as "Environment name". Add the publisher and you are good to go!
+
 
 ## Update your repo
 

